app/Models/Store/Order.php

```diff
@@ -283,10 +284,11 @@ private function newOrderItem(array $params)
 
         // FIXME: custom class stuff should probably not go in Order...
         if ($product->custom_class === 'supporter-tag') {
-            $targetUsername = $params['extraData']['username'];
-            $user = User::where('username', $targetUsername)->first();
-            // TODO: burst into flames if user is null
-            $params['extraData']['target_id'] = $user->user_id;
+            $targetId = $params['extraData']['target_id'];
+            $user = User::where('user_id', $targetId)->firstOrFail();
```

Comment:
This might be a bit unrelated, but are the target users checked to see if they're restricted/banned before allowing supporter tags to be gifted to them? I feel like that should probably be a thing?